File: backups/20250716_034306/test/mocks/mock_runtime_server.py
import http.server
import json
import os
import time
import random
from urllib.parse import urlparse

PORT = int(os.environ.get('PORT', 8082))
RESPONSE_DELAY_MS = int(os.environ.get('RESPONSE_DELAY_MS', 50))
ERROR_RATE = float(os.environ.get('ERROR_RATE', 0.1))

class MockRuntimeHandler(http.server.BaseHTTPRequestHandler):
    def do_GET(self):
        if self.path == '/health':
            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({'status': 'ok', 'runtime': 'python'}).encode())
        else:
            self.send_response(404)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({'error': 'Not found'}).encode())
    
    def do_POST(self):
        if self.path == '/execute':
            content_length = int(self.headers['Content-Length'])
            post_data = self.rfile.read(content_length)
            
            time.sleep(RESPONSE_DELAY_MS / 1000.0)
            
            if random.random() < ERROR_RATE:
                self.send_response(500)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                self.wfile.write(json.dumps({'error': 'Simulated runtime error'}).encode())
                return
            
            try:
                request_data = json.loads(post_data.decode('utf-8'))
                
                self.send_response(200)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                
                response = {
                    'result': {
                        'output': f"Executed {request_data['request_id']} with params {json.dumps(request_data['params'])}",
                        'language': 'python'
                    },
                    'execution_time_ms': RESPONSE_DELAY_MS,
                    'memory_usage_bytes': 1024 * 1024
                }
                
                self.wfile.write(json.dumps(response).encode())
            except Exception as e:
                self.send_response(400)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                self.wfile.write(json.dumps({'error': f'Invalid request data: {str(e)}'}).encode())
        else:
            self.send_response(404)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({'error': 'Not found'}).encode())

if __name__ == '__main__':
    server = http.server.HTTPServer(('', PORT), MockRuntimeHandler)
    print(f'Mock Python runtime server running on port {PORT}')
    server.serve_forever()

File: backups/20250715_222609/test/mocks/mock_runtime_server.js
const http = require('http');
const port = process.env.PORT || 8081;
const responseDelayMs = parseInt(process.env.RESPONSE_DELAY_MS || '50');
const errorRate = parseFloat(process.env.ERROR_RATE || '0.1');

const server = http.createServer((req, res) => {
  if (req.method === 'POST' && req.url === '/execute') {
    let body = '';
    req.on('data', chunk => {
      body += chunk.toString();
    });
    
    req.on('end', () => {
      setTimeout(() => {
        if (Math.random() < errorRate) {
          res.writeHead(500, { 'Content-Type': 'application/json' });
          res.end(JSON.stringify({ error: 'Simulated runtime error' }));
          return;
        }
        
        try {
          const requestData = JSON.parse(body);
          
          res.writeHead(200, { 'Content-Type': 'application/json' });
          res.end(JSON.stringify({
            result: { 
              output: `Executed ${requestData.request_id} with params ${JSON.stringify(requestData.params)}`,
              language: 'nodejs'
            },
            execution_time_ms: responseDelayMs,
            memory_usage_bytes: 1024 * 1024
          }));
        } catch (err) {
          res.writeHead(400, { 'Content-Type': 'application/json' });
          res.end(JSON.stringify({ error: 'Invalid request data' }));
        }
      }, responseDelayMs);
    });
  } else if (req.method === 'GET' && req.url === '/health') {
    res.writeHead(200, { 'Content-Type': 'application/json' });
    res.end(JSON.stringify({ status: 'ok', runtime: 'nodejs' }));
  } else {
    res.writeHead(404, { 'Content-Type': 'application/json' });
    res.end(JSON.stringify({ error: 'Not found' }));
  }
});

server.listen(port, () => {
  console.log(`Mock NodeJS runtime server running on port ${port}`);
});
